### PR TITLE
fix: Handle invalid git ignore globs gracefully

### PR DESCRIPTION
**Pull request by dustinbyrne**
If we fail to build a regex from an ignore glob, we now:
- log the glob
- filter the glob out and continue

### CODE REVIEW

**Comment by torchello** [comment]
Agree with @kgilpin. If don't need this exceptions for analytics (I guess we don't) we can just silently skip.

File: src/telemetry/properties/versionControlGit.ts [hunk 1]
```diff
@@ -3,17 +3,28 @@ import { PathLike, promises as fs } from 'fs';
...
     .split(/^/gm)
     .map((line) => line.trim().replace(/^\//, `${directory}/`))
     .filter((line) => line !== '')
-    .map((glob) => globToRegExp(glob, { flags: 'g' }));
+    .map((glob) => {
+      try {
+        return globToRegExp(glob, { flags: 'g' });
+      } catch {
+        const exception = new Error(`Invalid git ignore pattern: ${glob}`);
+        Telemetry.sendEvent(DEBUG_EXCEPTION, { exception });
```
Comment (kgilpin) [inline src/telemetry/properties/versionControlGit.ts:23]: Exception should be verified as being the expected type, and not sent to telemetry.

**Review by kgilpin — CHANGES_REQUESTED** [review]

**Comment by dustinbyrne** [comment]
I'd like to know what pattern is raising an exception, because I see no reason we should be failing here. With that data, we can issue a proper fix.

**Comment by dustinbyrne** [comment]
I'll address this in a different manner that replaces `globToRegExp` outright.

**Pull request closed without merging**